homeassistant/components/ondilo_ico/sensor.py

```diff
@@ -0,0 +1,185 @@
+"""Platform for sensor integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from ondilo import OndiloError
+
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+SENSOR_TYPES = {
+    "temperature": [
+        "Temperature",
+        TEMP_CELSIUS,
+        "mdi:thermometer",
+        DEVICE_CLASS_TEMPERATURE,
+    ],
+    "orp": ["Oxydo Reduction Potential", "mV", "mdi:pool", None],
+    "ph": ["pH", "", "mdi:pool", None],
+    "tds": ["TDS", CONCENTRATION_PARTS_PER_MILLION, "mdi:pool", None],
+    "battery": ["Battery", PERCENTAGE, "mdi:battery", DEVICE_CLASS_BATTERY],
+    "rssi": [
+        "RSSI",
+        PERCENTAGE,
+        "mdi:wifi-strength-2",
+        DEVICE_CLASS_SIGNAL_STRENGTH,
+    ],
+    "salt": ["Salt", "mg/L", "mdi:pool", None],
+}
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Ondilo ICO sensors."""
+
+    api = hass.data[DOMAIN][entry.entry_id]
+
+    def get_all_pool_data(pool):
+        """Add pool details and last measures to pool data."""
+        pool["ICO"] = api.get_ICO_details(pool["id"])
+        pool["sensors"] = api.get_last_pool_measures(pool["id"])
+
+        return pool
+
+    async def async_update_data():
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
+        so entities can quickly look up their data.
+        """
+        try:
+            pools = await hass.async_add_executor_job(api.get_pools)
+
+            return await asyncio.gather(
+                *[
+                    hass.async_add_executor_job(get_all_pool_data, pool)
+                    for pool in pools
+                ]
+            )
+
+        except OndiloError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=SCAN_INTERVAL,
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    entities = []
+    for poolidx, pool in enumerate(coordinator.data):
+        for sensor_idx, sensor in enumerate(pool["sensors"]):
+            if SENSOR_TYPES.get(sensor["data_type"], None) is not None:
```

Comment:
I believe the `None` argument to `get` is the default and can be omitted.  Also the "is not None" can be omitted. Another option is:

```
if sensor["data_type"] in SENSOR_TYPES:
```

Reply:
Exact... Not sure why I chose this way.... Updated.